app/components/hamburger/hamburger.tsx

```diff
@@ -0,0 +1,182 @@
+  SN_MAIN: "0x534e5f4d41494e", // Mainnet
+  SN_SEPOLIA: "0x534e5f5345504f4c4941", // Sepolia testnet
+};
+
+const CustomSelect: React.FC<{ theme: "dark" | "light" }> = ({ theme }) => {
+  const [selected, setSelected] = useState("Mainnet");
+  const [isOpen, setIsOpen] = useState(false);
+
+  const options = [
+    { name: "Mainnet", chainId: STARKNET_CHAINID.SN_MAIN },
+    { name: "Sepolia", chainId: STARKNET_CHAINID.SN_SEPOLIA },
+  ];
+
+  const { switchChain, error } = useSwitchChain({
+    params: {
+      chainId: STARKNET_CHAINID.SN_SEPOLIA,
+    },
+  });
+
+  const handleSelect = (option: { name: string; chainId: string }) => {
+    setSelected(option.name);
+    setIsOpen(false);
+
+    switchChain({ chainId: option.chainId });
+  };
+
+  return (
+    <div className="relative w-[52%] md:w-[316px]">
+      <div
+        className={`flex h-[50px] w-full cursor-pointer items-center justify-between rounded-xl px-4 ${
+          theme === "dark"
+            ? "border-[3px] border-grey-700 bg-[#2a2a2a] text-white"
+            : "border-[3px] border-[#eaeaea] bg-[#f0f0f0] text-black"
+        } md:font-[700]`}
+        onClick={() => setIsOpen(!isOpen)}
+      >
+        <span>{selected}</span>
+        <span>{isOpen ? "▲" : "▼"}</span>
+      </div>
+
+      {isOpen && (
+        <div className="absolute left-0 top-0 z-10 w-full rounded-xl border-[3px] border-[#eaeaea] bg-[#f0f0f0] px-4 py-2 text-black shadow-lg">
+          <div
+            className={`flex h-[50px] w-full cursor-pointer items-center justify-between px-4 ${
+              theme === "dark"
+                ? "border-b-[3px] border-grey-700 text-black"
+                : "border-[3px] border-[#eaeaea] bg-[#f0f0f0] text-black"
+            } md:font-[700]`}
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            <span>{selected}</span>
+            <span>{isOpen ? "▲" : "▼"}</span>
+          </div>
+          {options.map((option) => (
+            <div
+              key={option.name}
+              className={`mt-2 flex cursor-pointer items-center gap-2 rounded-md p-2 ${
+                selected === option.name ? "bg-white" : "hover:bg-gray-300"
+              }`}
+              onClick={() => handleSelect(option)}
+            >
+              {selected === option.name && (
+                <Check size={20} className="text-grey-800" />
+              )}
+              {option.name}
+            </div>
+          ))}
+        </div>
+      )}
+
+      {error && (
+        <p className="text-red-600">Error switching network: {error.message}</p>
+      )}
+    </div>
+  );
+};
+
+const Hamburger: React.FC<HamburgerProps> = ({ theme }) => {
+  const [open, setOpen] = useState(false);
+
+  return (
+    <div className="relative">
+      <button
+        className="z-10 flex h-8 w-8 cursor-pointer flex-col justify-around border-none bg-transparent p-0 focus:outline-none"
+        onClick={() => setOpen(!open)}
+      >
+        <div
```

Comment:
Please fix the translate in the hambutger icon. The open state looks a bit bent.